### ISSUE TITLE
Installation fails on OCP 4.3

### ISSUE THREAD

**Issue by csantanapr** [kabanero-io/kabanero-operator#404]
Environment:
- OCP 4.3
- Cloud Pak for Applications 4.0.0 (kabanero operator image `image: kabanero/kabanero-operator:0.3.4`)

Steps to reproduce:
- Run CP4Apps installer that installs kabanero operator

Results:
```
The InstallPlan status is 'Failed'. There might a problem that is preventing the InstallPlan installation from completing, or additional time might be required to complete.


Run the following commands to examine the status of the InstallPlan:
oc -n "kabanero" describe InstallPlan "install-htgqh"


Once the problem is resolved or the status of the InstallPlan becomes 'Complete', re-run the installer to complete the installation process.


oc -n "kabanero" describe InstallPlan "install-htgqh"
```

After running the command to describe the first failed components contains
```
Name:         install-htgqh
Namespace:    kabanero
Labels:       <none>
Annotations:  <none>
API Version:  operators.coreos.com/v1alpha1
Kind:         InstallPlan
Metadata:
  Creation Timestamp:  2020-01-21T20:13:46Z
  Generate Name:       install-
  Generation:          1
  Owner References:
    API Version:           operators.coreos.com/v1alpha1
    Block Owner Deletion:  false
    Controller:            false
    Kind:                  Subscription
    Name:                  ibm-cp-applications-kabanero
    UID:                   3a5eabd4-ded9-4071-9c69-60c34003ec87
  Resource Version:        266732
  Self Link:               /apis/operators.coreos.com/v1alpha1/namespaces/kabanero/installplans/install-htgqh
  UID:                     1bcef018-a7fa-47a8-8507-57e2660ad37f
Spec:
  Approval:  Automatic
  Approved:  true
  Cluster Service Version Names:
    kabanero-operator.v0.3.4
Status:
  Catalog Sources:
    ibm-cp-applications-catalog
  Conditions:
    Last Transition Time:  2020-01-21T20:13:47Z
    Last Update Time:      2020-01-21T20:13:47Z
    Message:               CustomResourceDefinition.apiextensions.k8s.io "collections.kabanero.io" is invalid: spec.versions: Invalid value: []apiextensions.CustomResourceDefinitionVersion{apiextensions.CustomResourceDefinitionVersion{Name:"v1alpha1", Served:true, Storage:true, Schema:(*apiextensions.CustomResourceValidation)(0xc02201c0a8), Subresources:(*apiextensions.CustomResourceSubresources)(nil), AdditionalPrinterColumns:[]apiextensions.CustomResourceColumnDefinition(nil)}}: per-version schemas may not all be set to identical values (top-level validation should be used instead)
    Reason:                InstallComponentFailed
    Status:                False
    Type:                  Installed
  Phase:                   Failed
  Plan:
    Resolving:  kabanero-operator.v0.3.4
    Resource:
      Group:     operators.coreos.com
      Kind:      ClusterServiceVersion
```

If you need the full result output ping me on Slack to share

**Comment by kaczyns** [comment]
@csantanapr Thanks for blazing the trail here :-) We understand that there are issues installing right now on OCP 4.3.  This specific issue relates to the CRD schema validation - when there is just one version of the schema, it is supposed to be declared at the "top level", outside of the `versions` array.  The version of `operator-sdk` that we used to generate this specific schema did not do this.  We have to make it compatible with Kube 1.16, then make sure that's still valid with Kube 1.14, then move on to the next problem.

**Comment by kaczyns** [comment]
@csantanapr starting with Kabanero 0.5.0 we are using operator-sdk 0.11.0 which generates CRDs which are compatible with Kube 1.16.  Would you mind trying to install Kabanero 0.5.0-rc.3?

**Comment by marikaj123** [comment]
@kaczyns - Greg raised concern at Ann's release delivery scrum meeting today.  Who can help with testing/reproducing this issue?

**Comment by kaczyns** [comment]
@marikaj123 there is no need to reproduce this issue.  We understand that Kabanero 0.5.0 is the first release of Kabanero that supports OCP 4.3.

It looks like the version of ICPA that Carlos is using, was based on Kabanero 0.3.4.  ICPA can move up to Kabanero 0.5.0 to fix their issue.  Or, if ICPA requires a new release of Kabanero 0.3.x that supports OCP 4.3, then ICPA can request that.

**Comment by kaczyns** [comment]
I've spoken to Carlos and Marika, and we agreed to close this issue since Kabanero release 0.5.0 supports OCP 4.3.  The ICPA matter will be handled separately in ICPA's repository.

**Comment by arobinson7532** [comment]
CP4Apps 4.0.0 uses Kabenero 0.4 release which has this issue.

**Comment by kaczyns** [comment]
@arobinson7532 Hi - I thought that ICPA and CP4Apps were the same thing.

**Comment by marikaj123** [comment]
@kaczyns - Was the issue resolved on Kabanero 0.5.0 and not on 0.4.0?  Are we saying we need to have Kabanero 0.4.0 to run on OCP 4.3?

**Comment by arobinson7532** [comment]
I agree with closing this issue and tracking it in the other github issue
